addons/overpressure/functions/fnc_firedEHBB.sqf

```diff
@@ -18,11 +18,11 @@
 //IGNORE_PRIVATE_WARNING ["_unit", "_weapon", "_muzzle", "_mode", "_ammo", "_magazine", "_projectile", "_vehicle", "_gunner", "_turret"];
 TRACE_10("firedEH:",_unit, _weapon, _muzzle, _mode, _ammo, _magazine, _projectile, _vehicle, _gunner, _turret);
 
-// Bake key name and check if it exists, call the caching function otherwise
-private _opData = (GVAR(cacheHash) getOrDefault [
-    (format ["%1#%2#%3", _weapon, _ammo, _magazine]),
-    ([_weapon, _ammo, _magazine] call FUNC(cacheOverPressureValues))
-]);
+// Check if key exists, call the caching function otherwise
+private _opData = GVAR(cacheHash) getOrDefault [
+    [_weapon, _ammo, _magazine],
+    [_weapon, _ammo, _magazine] call FUNC(cacheOverPressureValues)
```

Comment:
One point of note, I believe this function is always being called regardless of whether the key exists or not (due to the way SQF commands work, this array is fully constructed before `getOrDefault` does anything).

Reply:
Use something like this perhaps, or just move the check into `ace_overpressure_cacheOverPressureValues` and always do that call ?
```sqf
private _key = [_weapon, _ammo, _magazine];
private _opData = (
    if (_key in GVAR(cacheHash)) then {
        GVAR(cacheHash) get _key;
    } else {
        _key call FUNC(cacheOverPressureValues);
    }
);
```

Reply:
To avoid code duplication, probably better to add a function like `getOverPressureValues` that takes care of the caching details as well.

Reply:
Was just thinking adding this at the beginning of `ace_overpressure_cacheOverPressureValues`:
```sqf
private _key = [_weapon, _ammo, _magazine];
private _return = [];

// Return cached entry if it exists
if (_key in GVAR(cacheHash)) exitWith {
    _return = GVAR(cacheHash) get _key;
    TRACE_2("CacheHit",_key,_return);
    _return
};
```

And then just do this in the other functions:
```sqf
// Call the caching function to retrieve overpressure data
private _opData = [_weapon, _ammo, _magazine] call FUNC(cacheOverPressureValues);
```

Reply:
Standard implementation of a hash is such:
```sqf
private _value = GVAR(cache) getOrDefault [_key, nil];
if (isNil "_return") then {
    _value = _key call FUNC(expensiveFunction);
    GVAR(cache) set [_key, _value];
};
_value // return
```
A hash only makes sense if expensiveFunction is more expensive than traversing this snippet on average.

Reply:
So, I've added a `getOverPressureValues` function, which is basically just a wrapper for `cacheOverPressureValues`:
```sqf
private _values = GVAR(cache) get _this;

// Values not cached, call the caching function
if (isNil "_values") then {
    _values = _this call FUNC(cacheOverPressureValues);
};

_values
```

`private _opValues = [_weapon, _ammo, _magazine] call FUNC(getOverPressureValues);` then used in the EH functions.

Will this be satisfactory ?

Reply:
I think it's short enough you could just keep it inline in the 3 places it's used and not create a new func for it
```
private _opData = GVAR(cacheHash) get [_weapon, _ammo, _magazine];
if (isNil "_opData") then {
    _opData = [_weapon, _ammo, _magazine] call FUNC(cacheOverPressureValues);
};
```

Reply:
Personally I'd just rename the `chacheOverPressureValues` function to `getOverpressureValues` and then put the caching in there